proto2ros/include/proto2ros/conversions.hpp

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2024 Boston Dynamics AI Institute LLC. All rights reserved.
+
+/// This module provides basic conversion APIs, applicable to any proto2ros generated packages.
+
+#pragma once
+
+#include <stdexcept>
+
+#include <google/protobuf/any.pb.h>
+#include <google/protobuf/duration.pb.h>
+#include <google/protobuf/struct.pb.h>
+#include <google/protobuf/timestamp.pb.h>
+#include <google/protobuf/wrappers.pb.h>
+
+#include <builtin_interfaces/msg/duration.hpp>
+#include <builtin_interfaces/msg/time.hpp>
+#include <std_msgs/msg/float32.hpp>
+#include <std_msgs/msg/float64.hpp>
+#include <std_msgs/msg/int32.hpp>
+#include <std_msgs/msg/int64.hpp>
+#include <std_msgs/msg/u_int32.hpp>
+#include <std_msgs/msg/u_int64.hpp>
+#include <std_msgs/msg/bool.hpp>
+#include <std_msgs/msg/string.hpp>
+
+#include <proto2ros/msg/any.hpp>
+#include <proto2ros/msg/any_proto.hpp>
+#include <proto2ros/msg/bytes.hpp>
+#include <proto2ros/msg/list.hpp>
+#include <proto2ros/msg/struct.hpp>
+#include <proto2ros/msg/value.hpp>
+
+namespace proto2ros {
+namespace conversions {
+
+/// Unpacks a proto2ros/AnyProto ROS message into any Protobuf message.
+///
+/// \throws std::runtime_error if the given ROS message cannot be unpacked onto the given Protobuf message.
+template <typename T>
+void convert(const proto2ros::msg::AnyProto& ros_msg, T* proto_msg) {
```

Comment:
Thoughts about using references instead of pointers? Unless you plan to check for null everywhere I think they are safer.

Reply:
Ah... yeah with `mutable` fields returning pointers instead of references I think this is fine. I don't like the `OutputT convert(const InputT&)` style because that triggers a copy/move instead of using the existing field.